### PR TITLE
Fix flavor id method to return valid id.

### PR DESCRIPTION
**Pull request by cryptickp**
If get_flavor_id gets id or name checks and returns valid id

CBD-4366 #fixes.

### CODE REVIEW

**Comment by cryptickp** [comment]
@rs-randallburt @PaulMontgomery : Minor fix.

File: rackspace/cbd_client.py [hunk 1]
```diff
@@ -66,16 +66,19 @@ def get_flavor_id(self, flavor):
...
         except LavaError as exc:
             LOG.info("Unable to read CBD flavor list", exc_info=exc)
             raise
         for bigdata_flavor in flavor_list:
-            if bigdata_flavor.name == flavor:
-                return bigdata_flavor.id
-        LOG.info("Unable to find CBD flavor %s", flavor)
-        raise exception.FlavorMissing(flavor_id=flavor)
+            if bigdata_flavor.name == flavor or bigdata_flavor.id == flavor:
+                flavor_id = bigdata_flavor.id
```
Comment (rs-randallburt) [inline rackspace/cbd_client.py]: Might want to break here to save a cycle or two.

**Comment by rs-randallburt** [comment]
:+1:

**Comment by cryptickp** [comment]
Merging.

**Pull request merged**